armonik/outputs.tf

```
@@ -1,3 +1,52 @@
+locals {
+  # Local variables to determine endpoint details
+  tls_enabled  = var.ingress != null ? var.ingress.tls : false
+  mtls_enabled = var.ingress != null ? var.ingress.mtls : false
+
+  # Determine endpoint and protocol
+  grpc_protocol = local.tls_enabled ? "https" : "http"
+  endpoint_url = var.ingress != null ? (
+    "${local.grpc_protocol}://${module.ingress_endpoint.ip}:${var.ingress.grpc_port}"
+  ) : "${module.control_plane_endpoint.ip}:${var.control_plane.port}"
+
+  # Base configuration map with endpoint
+  armonik_config_map = {
+    endpoint = local.endpoint_url
+  }
+
+  # Add certificates based on TLS/mTLS configuration
+  armonik_config_with_certs = local.tls_enabled ? (
+    local.mtls_enabled && var.ingress.generate_client_cert ? merge(
+      local.armonik_config_map,
+      {
+        certificate_authority = abspath("${path.root}/generated/certificates/ingress/ca.crt")
+        client_certificate    = abspath("${path.root}/generated/certificates/ingress/client.submitter.crt")
+        client_key            = abspath("${path.root}/generated/certificates/ingress/client.submitter.key")
+      }
+      ) : merge(
+      local.armonik_config_map,
+      {
+        certificate_authority = abspath("${path.root}/generated/certificates/ingress/ca.crt")
+      }
+    )
+  ) : local.armonik_config_map
+
+  armonik_config_yaml = "# ArmoniK Connection Configuration\n---\n${yamlencode(local.armonik_config_with_certs)}"
+}
+
+# Create the configuration file
+resource "local_file" "armonik_config" {
+  content         = local.armonik_config_yaml
+  filename        = "${path.root}/generated/armonik-cli.yaml"
+  file_permission = "0644"
+}
+
+output "armonik_config_file" {
+  description = "Path to the generated ArmoniK configuration file"
+  value       = local_file.armonik_config.filename != null ? abspath(local_file.armonik_config.filename) : null
```

`local_file.armonik_config.filename` cannot be null

```suggestion
  value       = abspath(local_file.armonik_config.filename)
```